wait/log.go

```diff
@@ -24,6 +41,18 @@ type LogStrategy struct {
 	IsRegexp     bool
 	Occurrence   int
 	PollInterval time.Duration
+
+	// check is the function that will be called to check if the log entry is present.
+	check func([]byte) error
+
+	// submatchCallback is a callback that will be called with the sub matches of the regexp.
+	submatchCallback func(pattern string, matches [][][]byte) error
+
+	// re is the optional compiled regexp.
+	re *regexp.Regexp
+
+	// log byte slice version of [LogStrategy.Log] used for count checks.
+	log []byte
```

Comment:
bug: we need to document these new fields, and relative methods, in the docs site.

Reply:
Just the new method, internal fields shouldn't be documented.

Reply:
Sorry, I meant the new exposed methods